nuxeo-core/nuxeo-core-test/src/test/java/org/nuxeo/ecm/core/TestSQLRepositoryQuery.java

```diff
@@ -3853,4 +3855,100 @@ public void testQueryDefaultValueInComplex() {
         assertEquals(1, dml.size());
     }
 
+    @Test
+    public void testIsRecord() {
+        String queryRecords = "SELECT * FROM MyDocType WHERE ecm:isRecord = 1";
+        String queryNotRecord = "SELECT * FROM MyDocType WHERE ecm:isRecord = 0";
+
+        // no record present initially
+        DocumentModelList dml ;
+        dml = session.query(queryRecords);
+        assertEquals(0, dml.size());
+        dml = session.query(queryNotRecord) ;
+        assertEquals(0, dml.size());
+
+        // create a document
+        DocumentModel doc = session.createDocumentModel("/", "doc", "MyDocType");
+        doc = session.createDocument(doc);
+        session.save();
+
+        // still no record
+        dml = session.query(queryRecords);
+        assertEquals(0, dml.size());
+        dml = session.query(queryNotRecord) ;
+        assertEquals(1, dml.size()); // 1 doc that isn't a record
+
+        // make a record
+        session.makeRecord(doc.getRef());
+        session.save();
+
+        // check that we can find the record
+        dml = session.query(queryRecords);
+        assertEquals(1, dml.size());
+        assertEquals(doc.getId(), dml.get(0).getId());
+        dml = session.query(queryNotRecord) ;
+        assertEquals(0, dml.size());
+    }
+
+    @Test
+    public void testRetainUntil() {
+        Calendar fiveSeconds = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
+        fiveSeconds.add(Calendar.SECOND, 5);
+        String query = String.format("SELECT * FROM MyDocType WHERE ecm:retainUntil > TIMESTAMP '%s'",
+                DateParser.formatW3CDateTime(fiveSeconds));
+        DocumentModelList dml = session.query(query);
```

Comment:
i am wondering what if we use `java.time` -> `ZonedDateTime.now(ZoneId.of("UTC")).plusSeconds(5).format(DateTimeFormatter.ISO_INSTANT);`

Reply:
We could yes, it's just copy/pasted from much older code that predates `java.time`. Not worth changing in a test for now.